### PR TITLE
feat(ctb): Add tests for `Bytes` library

### PR DESCRIPTION
**Pull request by clabby**
# Overview

Adds tests for the functions in the Bedrock contracts' `Bytes` lib.

### CODE REVIEW

**Comment by clabby** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4426** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/monorepo/4426" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/monorepo/4426?utm_source=stack-comment).

File: packages/contracts-bedrock/contracts/test/Bytes.t.sol [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+    /// @dev Tests that, when given an input bytes array of length `n`,
+    /// the `slice` function will always revert if `_start + _length > n`.
+    function testFuzz_slice_outOfBounds_reverts(
+        bytes memory _input,
+        uint256 _start,
+        uint256 _length
+    ) public {
+        // We want a valid start index and a length that will not overflow.
+        vm.assume(_start < _input.length && _length < type(uint256).max - 31);
```
Comment (maurelian) [inline packages/contracts-bedrock/contracts/test/Bytes.t.sol:76]: could we use `bound()` here instead to increase the number of test cases that actually run?

Comment (maurelian) [reply packages/contracts-bedrock/contracts/test/Bytes.t.sol:76]: Same for other locations using vm.assume()

File: packages/contracts-bedrock/contracts/test/Bytes.t.sol [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    /// a zero length array.
+    function test_toNibbles_zeroLengthInput_works() public {
+        bytes memory input = hex"";
+        bytes memory expected = hex"";
+        bytes memory actual = Bytes.toNibbles(input);
+
+        assertEq(input.length, 0);
+        assertEq(expected.length, 0);
+        assertEq(actual.length, 0);
+        assertEq(Bytes.toNibbles(input), expected);
```
Comment (maurelian) [inline packages/contracts-bedrock/contracts/test/Bytes.t.sol:153]: nit, why not just: 
```suggestion
        assertEq(actual, expected);
```
?

Comment (clabby) [reply packages/contracts-bedrock/contracts/test/Bytes.t.sol:153]: Ah whoops - copy / paste error

File: packages/contracts-bedrock/contracts/test/Bytes.t.sol (hunk 1 above)
Comment (clabby) [reply packages/contracts-bedrock/contracts/test/Bytes.t.sol:76]: Will do 👍

**Review by tynes — APPROVED** [review]

**Pull request merged**